docs: improve docs for amd (#9674)
web-infra-dev/rspack: website/docs/en/config/other-options.mdx

## amd

<ApiMeta addedVersion="1.3.0" />

<PropertyType
  type="false | object"
  defaultValueList={[{ defaultValue: 'false' }]}
/>

:::info
Different from Webpack, in Webpack, the default value of `amd` is `{}`, and the dependency analysis of AMD modules is enabled by default. However, since the usage of AMD modules is gradually decreasing at present, the default value of `amd` in Rspack is `false`, and the dependency analysis of AMD modules is disabled by default. If your project requires it, you can enable this function by yourself.
:::

> It supports the dependency analysis of AMD modules. Some old libraries are written according to the AMD specification. You can enable this configuration to be compatible with these libraries.

Reviewer: ```suggestion
Enable this option to support dependency analysis for AMD modules, which is helpful for compatibility with some older libraries written according to the AMD specification.
```
